### PR TITLE
Fix manual locale/language set problem, SDK29  / Q+

### PR DESCRIPTION
**Pull request by norkator**
* Fixing locale problems.

### CODE REVIEW

**Comment by developerfromjokela** [comment]
Is it working? @norkator have you tested this?

**Comment by norkator** [comment]
> Is it working? @norkator have you tested this?

Work in progress, I open draft PR's to connect issues to it.

**Comment by norkator** [comment]
Weird that main activity works now but other activities doesn't, trying to find what is causing that.

**Comment by developerfromjokela** [comment]
Is the change also in Application class?

**Comment by norkator** [comment]
> Is the change also in Application class?

If you mean "PaketinSeurantaApp" class -> no, if I put it there, then whole app is again same locale as OS.

**Comment by norkator** [comment]
I need to still later do more testing with different Android versions to see should I exclude this setApplicationLanguage from some specific OS versions.

**Comment by developerfromjokela** [comment]
Can you summarize what did you to to get it working? 😃

**Comment by norkator** [comment]
> Can you summarize what did you to to get it working? 😃

Added updateResourcesLocaleQ function on localeutils, after this MainActivity started to work with set language at settings.

Then to get rest of the activities to work:

Dumped localeUtils.setApplicationLanguage(this); to every onCreate after super.onCreate(savedInstanceState);

protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        localeUtils.setApplicationLanguage(this);

setApplicationLanguage function is found from stackoverflow 🤣  

This feels like big hack again.

**Comment by norkator** [comment]
@developerfromjokela by the way, I have been digging that preference system you changed earlier, very great work with this so got rid of that old fnp material preference junk.

**Comment by developerfromjokela** [comment]
> > Can you summarize what did you to to get it working? smiley
> 
> Added updateResourcesLocaleQ function on localeutils, after this MainActivity started to work with set language at settings.
> 
> Then to get rest of the activities to work:
> 
> Dumped localeUtils.setApplicationLanguage(this); to every onCreate after super.onCreate(savedInstanceState);
> 
> protected void onCreate(Bundle savedInstanceState) {
> super.onCreate(savedInstanceState);
> localeUtils.setApplicationLanguage(this);
> 
> setApplicationLanguage function is found from stackoverflow rofl
> 
> This feels like big hack again.

Wouldn't it be better to make one activity class containing all locale utilities and extending others to that class?
Saving some code from duplicating is always great

**Comment by norkator** [comment]
> > > Can you summarize what did you to to get it working? smiley
> > 
> > 
> > Added updateResourcesLocaleQ function on localeutils, after this MainActivity started to work with set language at settings.
> > Then to get rest of the activities to work:
> > Dumped localeUtils.setApplicationLanguage(this); to every onCreate after super.onCreate(savedInstanceState);
> > protected void onCreate(Bundle savedInstanceState) {
> > super.onCreate(savedInstanceState);
> > localeUtils.setApplicationLanguage(this);
> > setApplicationLanguage function is found from stackoverflow rofl
> > This feels like big hack again.
> 
> Wouldn't it be better to make one activity class containing all locale utilities and extending others to that class?
> Saving some code from duplicating is always great

Yeah, true.

**Comment by norkator** [comment]
@developerfromjokela I made class LocaleClass which extends AppCompatActivity having attachBaseContext as usual then onCreate because localeUtils.setApplicationLanguage(this); needs to be run after super.onCreate(savedInstanceState); buut whole thing now crashing. I leave this like it is now, someone else can improve if wants.

**Comment by developerfromjokela** [comment]
I confirm now it works perfectly. Language is right all the time

**Comment by norkator** [comment]
> I confirm now it works perfectly. Language is right all the time

Great, what about that class extending idea, can you show some light how to do it right?
